File: apps/pos/src/services/product.service.ts
import api from './api';

export interface Category {
  id: string;
  name: string;
  nameRu?: string;
  slug: string;
  image?: string;
  sortOrder: number;
  isActive: boolean;
}

export interface Product {
  id: string;
  name: string;
  nameRu?: string;
  price: number;
  costPrice?: number;
  categoryId: string;
  cookingTime?: number;
  image?: string;
  barcode?: string;
  isActive: boolean;
}

export const productService = {
  getAll: async (params?: { categoryId?: string; search?: string }): Promise<Product[]> => {
    const { data: response } = await api.get('/products', { params: { ...params, limit: 200 } });
    return response.data?.data || response.data || [];
  },

  getById: async (id: string): Promise<Product> => {
    const { data: response } = await api.get(`/products/${id}`);
    return response.data;
  },

  getByBarcode: async (barcode: string): Promise<Product> => {
    const { data: response } = await api.get(`/products/barcode/${barcode}`);
    return response.data;
  },
};

export const categoryService = {
  getAll: async (): Promise<Category[]> => {
    const { data: response } = await api.get('/categories');
    return response.data || [];
  },
};

File: apps/waiter_flutter/lib/services/product_service.dart
import '../core/api.dart';
import '../models/product.dart';

class ProductService {
  Future<List<Product>> getAll({String? categoryId}) async {
    final Map<String, dynamic> queryParams = {};
    if (categoryId != null) queryParams['categoryId'] = categoryId;

    final response = await dio.get(
      '/products',
      queryParameters: queryParams.isNotEmpty ? queryParams : null,
    );
    final data = response.data;
    final Map<String, dynamic> body = data is Map<String, dynamic> ? data : {};
    final List<dynamic> list = body['data'] as List<dynamic>? ??
        body['products'] as List<dynamic>? ??
        (data is List ? data : []);
    return list.map((e) => Product.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<List<Category>> getCategories() async {
    final response = await dio.get('/categories');
    final data = response.data;
    final Map<String, dynamic> body = data is Map<String, dynamic> ? data : {};
    final List<dynamic> list = body['data'] as List<dynamic>? ??
        body['categories'] as List<dynamic>? ??
        (data is List ? data : []);
    return list.map((e) => Category.fromJson(e as Map<String, dynamic>)).toList();
  }
}
